runtime/centrifuge/src/xcm.rs

```diff
@@ -0,0 +1,365 @@
+use frame_support::sp_std::marker::PhantomData;
+use frame_support::traits::fungibles;
+pub use frame_support::{
+	parameter_types,
+	traits::{Contains, Everything, Get, Nothing},
+	weights::Weight,
+};
+use orml_asset_registry::{AssetRegistryTrader, FixedRateAssetRegistryTrader};
+use orml_traits::{location::AbsoluteReserveProvider, parameter_type_with_key, MultiCurrency};
+use orml_xcm_support::MultiNativeAsset;
+use pallet_xcm::XcmPassthrough;
+use polkadot_parachain::primitives::Sibling;
+use sp_runtime::traits::{Convert, Zero};
+use xcm::latest::prelude::*;
+use xcm_builder::{
+	AccountId32Aliases, AllowKnownQueryResponses, AllowSubscriptionsFrom,
+	AllowTopLevelPaidExecutionFrom, ConvertedConcreteAssetId, EnsureXcmOrigin, FixedRateOfFungible,
+	FixedWeightBounds, FungiblesAdapter, LocationInverter, ParentIsPreset, RelayChainAsNative,
+	SiblingParachainAsNative, SiblingParachainConvertsVia, SignedAccountId32AsNative,
+	SignedToAccountId32, SovereignSignedViaLocation, TakeRevenue, TakeWeightCredit,
+};
+use xcm_executor::{traits::JustTry, XcmExecutor};
+
+use runtime_common::{
+	decimals, parachains,
+	xcm::FixedConversionRateProvider,
+	xcm_fees::{default_per_second, native_per_second},
+	CurrencyId,
+};
+
+/// The main XCM config
+/// This is where we configure the core of our XCM integrations: how tokens are transferred,
+/// how fees are calculated, what barriers we impose on incoming XCM messages, etc.
+pub struct XcmConfig;
+impl xcm_executor::Config for XcmConfig {
+	type Call = Call;
+	type XcmSender = XcmRouter;
+	// How to withdraw and deposit an asset.
+	type AssetTransactor = FungiblesTransactor;
+	type OriginConverter = XcmOriginToTransactDispatchOrigin;
+	type IsReserve = MultiNativeAsset<AbsoluteReserveProvider>;
+	type IsTeleporter = ();
+	type LocationInverter = LocationInverter<Ancestry>;
+	type Barrier = Barrier;
+	type Weigher = FixedWeightBounds<UnitWeightCost, Call, MaxInstructions>;
+	type Trader = Trader;
+	type ResponseHandler = PolkadotXcm;
+	type AssetTrap = PolkadotXcm;
+	type AssetClaims = PolkadotXcm;
+	type SubscriptionService = PolkadotXcm;
+}
+
+/// Trader - The means of purchasing weight credit for XCM execution.
+/// We need to ensure we have at least one rule per token we want to handle or else
+/// the xcm executor won't know how to charge fees for a transfer of said token.
+pub type Trader = (
+	FixedRateOfFungible<CanonicalCfgPerSecond, ToTreasury>,
+	FixedRateOfFungible<CfgPerSecond, ToTreasury>,
+	FixedRateOfFungible<AUSDPerSecond, ToTreasury>,
+	AssetRegistryTrader<
+		FixedRateAssetRegistryTrader<FixedConversionRateProvider<OrmlAssetRegistry>>,
+		ToTreasury,
+	>,
+);
+
+parameter_types! {
+	// Canonical location: https://github.com/paritytech/polkadot/pull/4470
+	pub CanonicalCfgPerSecond: (AssetId, u128) = (
+		MultiLocation::new(
+			0,
+			X1(GeneralKey(parachains::polkadot::centrifuge::CFG_KEY.to_vec())),
+		).into(),
+		native_per_second(),
+	);
+
+	pub CfgPerSecond: (AssetId, u128) = (
```

Comment:
If you prefer that's ok for me; the reason I wasn't consistent here is that I can't make it `Ausd`, it would need to be `AUsd` which is weird; and for example `AIRPerSecond` hurts my eyes :P I'm fine either way so let me know how you would prefer to have it

Reply:
I really have no preference. It all depends of what is the common style used in these cases (which I don't know yet). But using the same nomenclature in all cases will improve future readings. Anyway, it's a silly minor thing, so if for this letting as it is clear enough, for me is ok. 😃